internal/cmd/asyncapi/command_import.go

```diff
@@ -0,0 +1,402 @@
+package asyncapi
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/antihax/optional"
+	kafkarestv3 "github.com/confluentinc/ccloud-sdk-go-v2/kafkarest/v3"
+	srsdk "github.com/confluentinc/schema-registry-sdk-go"
+	"github.com/go-yaml/yaml"
+	"github.com/iancoleman/strcase"
+	"github.com/spf13/cobra"
+	spec2 "github.com/swaggest/go-asyncapi/spec-2.4.0"
+	yaml3 "k8s.io/apimachinery/pkg/util/yaml"
+
+	"github.com/confluentinc/cli/internal/pkg/ccloudv2"
+	pcmd "github.com/confluentinc/cli/internal/pkg/cmd"
+	"github.com/confluentinc/cli/internal/pkg/errors"
+	"github.com/confluentinc/cli/internal/pkg/kafkarest"
+	"github.com/confluentinc/cli/internal/pkg/log"
+	"github.com/confluentinc/cli/internal/pkg/output"
+	"github.com/confluentinc/cli/internal/pkg/utils"
+)
+
+type flagsImport struct {
+	overwrite               bool
+	kafkaApiKey             string
+	schemaRegistryApiKey    string
+	schemaRegistryApiSecret string
+}
+
+type BindingsXConfigs struct {
+	CleanupPolicy                  string `yaml:"cleanup.policy"`
+	DeleteRetentionMs              int    `yaml:"delete.retention.ms"`
+	ConfluentValueSchemaValidation string `yaml:"confluent.value.schema.validation"`
+}
+
+type KafkaBinding struct {
+	XPartitions int              `yaml:"x-partitions"`
+	XReplicas   int              `yaml:"x-replicas"`
+	XConfigs    BindingsXConfigs `yaml:"x-configs"`
+}
+
+type Message struct {
+	SchemaFormat string      `yaml:"schemaFormat"`
+	ContentType  string      `yaml:"contentType"`
+	Payload      interface{} `yaml:"payload"`
+	Name         string      `yaml:"name"`
+	Tags         []spec2.Tag `yaml:"tags"`
+}
+
+type Spec struct {
+	Asyncapi   string           `yaml:"asyncapi"`
+	Info       spec2.Info       `yaml:"info"`
+	Servers    spec2.Server     `yaml:"servers"`
+	Channels   map[string]Topic `yaml:"channels"`
+	Components Components       `yaml:"components"`
+}
+
+type Components struct {
+	Messages        map[string]Message `yaml:"messages"`
+	SecuritySchemes interface{}        `yaml:"securitySchemes"`
+}
+
+type Security struct {
+	ConfluentBroker         []interface{} `yaml:"confluentBroker"`
+	ConfluentSchemaRegistry []interface{} `yaml:"confluentSchemaRegistry"`
+}
+
+type MessageRef struct {
+	Ref string `yaml:"$ref"`
+}
+
+type Operation struct {
+	OperationID string                        `yaml:"operationId"`
+	TopicTags   []spec2.Tag                   `yaml:"tags"`
+	OpBindings  spec2.OperationBindingsObject `yaml:"bindings"`
+	Message     MessageRef                    `yaml:"message"`
+}
+
+type Topic struct {
+	Publish               Operation     `yaml:"publish"`
+	Subscribe             Operation     `yaml:"subscribe"`
+	XMessageCompatibility string        `yaml:"x-messageCompatibility"`
+	TopicBinding          TopicBindings `yaml:"bindings"`
```

Comment:
Out of curiosity, why are some fields prefixed with "x-" but others are not?

Reply:
In the AsyncAPI specification, custom fields can be added by prefixing `x-` to it. The struct naming has been done on the basis of their yaml counterparts.